juspay/hyperswitch
# feat: add support for merchant to pass public key and ttl for encrypting payload

## Type of Change



## Description

This PR adds support for merchant to pass his public key as well as ttl using update endpoint of business profile. The config passed is stored in business profile table as JSONB.

### Additional Changes





## Motivation and Context



## How did you test it?

1. Create a business profile for a merchant (Default one which gets created during merchant account creation works too)
2. Use update endpoint of business profile to pass config.
```
curl --location 'http://localhost:8080/account/:merchant_id/business_profile/:profile_id' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data '{
    "extended_card_info_config": {
        "public_key": "pub_key_1",
        "ttl_in_secs": 60
    }
}'
```
![image](https://github.com/juspay/hyperswitch/assets/70657455/67787f0e-724c-4bed-ae37-a2636093e481)

3. For verifying, u could check the db if the config is stored in `extended_card_info_config` column in `business_profile` table using below query
```
SELECT extended_card_info_config FROM business_profile WHERE profile_id = 'profile_id';
```

## Checklist
